index.js

```diff
@@ -3,30 +3,22 @@ var path = require('path');
 var pathExists = require('path-exists');
 var Promise = require('pinkie-promise');
 
-function splitPath(x) {
-	return path.resolve(x || '').split(path.sep);
-}
-
-function join(parts, filename) {
-	return path.resolve(parts.join(path.sep) + path.sep, filename);
-}
-
 module.exports = function (filename, opts) {
 	opts = opts || {};
 
-	var parts = splitPath(opts.cwd);
+	var dir = path.resolve(opts.cwd || '');
 
 	return new Promise(function (resolve) {
 		(function find() {
-			var fp = join(parts, filename);
-
+			var fp = path.join(dir, filename);
 			pathExists(fp).then(function (exists) {
 				if (exists) {
 					resolve(fp);
-				} else if (parts.pop()) {
-					find();
-				} else {
+				} else if (dir === path.sep) {
```

Comment:
I was wondering about that. Seems like Node doesn't have an API to get the root of a path. I guess I would `substring()` to the first `path.sep` from the resolved `cwd`.